Sample-01/src/views/ExternalApi.js

```diff
@@ -0,0 +1,144 @@
+    showResult: false,
+    apiMessage: "",
+    error: null,
+  });
+
+  const {
+    getAccessTokenSilently,
+    loginWithPopup,
+    getAccessTokenWithPopup,
+  } = useAuth0();
+
+  const handleConsent = async () => {
+    try {
+      await getAccessTokenWithPopup();
+      setState({
+        ...state,
+        error: null,
+      });
+    } catch (error) {
+      setState({
+        ...state,
+        error: error.error,
+      });
+    }
+
+    await callApi();
+  };
+
+  const handleLoginAgain = async () => {
+    try {
+      await loginWithPopup();
+      setState({
+        ...state,
+        error: null,
+      });
+    } catch (error) {
+      setState({
+        ...state,
+        error: error.error,
+      });
+    }
+
+    await callApi();
+  };
+
+  const callApi = async () => {
+    try {
+      const token = await getAccessTokenSilently();
+
+      const response = await fetch(`${apiOrigin}/api/external`, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+
+      const responseData = await response.json();
+
+      setState({
+        ...state,
+        showResult: true,
+        apiMessage: responseData,
+      });
+    } catch (error) {
+      setState({
+        ...state,
+        error: error.error,
+      });
+    }
+  };
+
+  const handle = (e, fn) => {
+    e.preventDefault();
+    fn();
+  };
+
+  return (
+    <>
+      <div className="mb-5">
+        {state.error === "consent_required" && (
+          <Alert color="warning">
+            You need to{" "}
+            <a
+              href="#/"
+              class="alert-link"
+              onClick={(e) => handle(e, handleConsent)}
+            >
+              consent to get access to users api
+            </a>
+          </Alert>
+        )}
+
+        {state.error === "login_required" && (
+          <Alert color="warning">
+            You need to{" "}
+            <a
+              href="#/"
+              class="alert-link"
+              onClick={(e) => handle(e, handleLoginAgain)}
+            >
+              log in again
+            </a>
+          </Alert>
+        )}
+
+        <h1>External API</h1>
+        <p>
+          Ping an external API by clicking the button below. This will call the
+          external API using an access token, and the API will validate it using
+          the API's audience value.
+        </p>
+
+        <Button color="primary" className="mt-5" onClick={callApi}>
+          Ping API
+        </Button>
+      </div>
+
+      <div className="result-block-container">
+        {state.showResult && (
+          <div className="result-block" data-testid="api-result">
+            <h6 className="muted">Result</h6>
+            <Highlight>
+              <span>{JSON.stringify(state.apiMessage, null, 2)}</span>
+            </Highlight>
+          </div>
+        )}
+      </div>
+    </>
+  );
+};
+
+export default withAuthenticationRequired(ExternalApiComponent, () => (
+  <Loading />
```

Comment:
fyi, we'll need to change this to
`withAuthenticationRequired(ExternalApiComponent, { onRedirecting: () => <Loading/>});`
for 1.0.0